### PR TITLE
feat(api): add enable type validation

### PR DESCRIPTION
**Pull request by djabarovgeorge**
### What changed? Why was the change needed?
Fix - CastError: Cast to Boolean failed for value "" (type string) at path "channels.sms" because of "CastError"

### Screenshots

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: apps/api/src/app/shared/dtos/channel-preference.ts
Comment (djabarovgeorge) [inline apps/api/src/app/shared/dtos/channel-preference.ts:1]: added missing validation on our dto.

File: apps/api/src/app/subscribers/e2e/update-preference.e2e.ts [hunk 1]
```diff
@@ -31,7 +31,7 @@ describe('Update Subscribers preferences - /subscribers/:subscriberId/preference
 
     try {
       const response = await updatePreference(updateDataEmailFalse as any, session, template._id);
-      expect(response).to.not.be;
```
Comment (djabarovgeorge) [inline apps/api/src/app/subscribers/e2e/update-preference.e2e.ts:34]: we were missing the 'ok' in order test in order to trigger failure if the test is falsely passed.

File: apps/api/src/app/widgets/dtos/update-subscriber-preference-request.dto.ts [hunk 2]
```diff
@@ -9,6 +10,8 @@ export class UpdateSubscriberPreferenceRequestDto {
     description: 'The subscriber preferences for every ChannelTypeEnum for the workflow assigned.',
   })
   @ValidateNested()
+  @Type(() => ChannelPreference)
```
Comment (djabarovgeorge) [inline apps/api/src/app/widgets/dtos/update-subscriber-preference-request.dto.ts:13]: needed in order to execute `@ValidateNested()`

**Review by ainouzgali — APPROVED** [review]
🤩

**Pull request merged**